test/algorithms/envelope_on_spheroid.cpp

```diff
@@ -38,125 +41,53 @@
-{
-    if (BOOST_GEOMETRY_CONDITION((boost::is_same<Units, bg::degree>::value)))
-    {
-        return "degrees";
-    }
-    return "radians";
-}
-
-template <typename Units>
-struct other_system_info
-{
-    typedef bg::degree units;
-    typedef bg::cs::spherical_equatorial<units> type;
-
-    template <typename T>
-    static inline T convert(T const& value)
-    {
-        return value * bg::math::r2d<T>();
-    }
-};
-
-template <>
-struct other_system_info<bg::degree>
-{
-    typedef bg::radian units;
-    typedef bg::cs::spherical_equatorial<units> type;
-
-    template <typename T>
-    static inline T convert(T const& value)
-    {
-        return value * bg::math::d2r<T>();
-    }
-};
-
-
-class equals_with_tolerance
+template <typename MBR>
+class envelope_on_spheroid_basic_tester
 {
 private:
-    double m_tolerance;
-
-    template <typename T>
-    static inline T const& get_max(T const& a, T const& b, T const& c)
+    template
+    <
+        typename Geometry,
+        typename Tag = typename bg::tag<Geometry>::type
+    >
+    struct write_geometry
     {
-        return (std::max)((std::max)(a, b), c);
-    }
+        template <typename OutputStream>
+        static inline OutputStream& apply(OutputStream& os,
+                                          Geometry const& geometry)
+        {
+            os << bg::wkt(geometry);
+            return os;
+        }
+    };
 
-    template <typename T>
-    static inline bool check_close(T const& a, T const& b, double tol)
+    template <typename Segment>
+    struct write_geometry<Segment, bg::segment_tag>
     {
-        return (a == b)
-            || (std::abs(a - b) <= tol * get_max(std::abs(a), std::abs(b), 1.0));
-    }
-
-public:
-    equals_with_tolerance(double tolerance) : m_tolerance(tolerance) {}
+        template <typename OutputStream>
+        static inline OutputStream& apply(OutputStream& os,
+                                          Segment const& segment)
+        {
+            os << "SEGMENT" << bg::dsv(segment);
+            return os;
+        }
+    };
 
-    template <typename T>
-    inline bool operator()(T const& value1, T const& value2) const
+    template <typename Box>
+    struct write_geometry<Box, bg::box_tag>
     {
-        return check_close(value1, value2, m_tolerance);
-    }
-};
-
-
-template <typename Box1, typename Box2>
-inline bool box_equals(Box1 const& box1, Box2 const& box2, double tol)
-{
-    equals_with_tolerance equals(tol);
-
-    return equals(bg::get<0, 0>(box1), bg::get<0, 0>(box2))
-        && equals(bg::get<0, 1>(box1), bg::get<0, 1>(box2))
-        && equals(bg::get<1, 0>(box1), bg::get<1, 0>(box2))
-        && equals(bg::get<1, 1>(box1), bg::get<1, 1>(box2));
-}
-
+        template <typename OutputStream>
+        static inline OutputStream& apply(OutputStream& os,
+                                          Box const& box)
+        {
+            os << "BOX" << bg::dsv(box);
+            return os;
+        }
+    };
```

Comment:
`bg::wkt` outputs boxes as polygons. I do not particularly like this for two reasons:

1. In debug mode, or when an error occurs, I prefer to see the word `BOX` instead of `POLYGON` in the output (it makes it easier for me to register that the involved geometry is a box).
2. In the spherical equatorial system, a box is not really a polygon (whose segments are geodesics on the sphere), but rather a min and a max point indicating the minimum and maximum longitude and latitude coordinates of a geometry.